analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,672 @@
+# use the jaccard matrices to help guide assignments
+cluster_umap_df <- cluster_classification_df |>
+  dplyr::left_join(umap_df, by = c("library_id", "barcodes")) |>
+  dplyr::filter(nn_param == "k_25") |>
+  dplyr::mutate(
+    cluster_celltype = dplyr::case_when(
+      library_id == "SCPCL000822" & cluster %in% c(1, 2, 4, 5, 9) ~ "tumor",
+      library_id == "SCPCL000822" & cluster == 7 ~ "endothelial cell",
+      # cluster 3 has predominantly chondrocytes and has MSC marker gene expression
+      library_id == "SCPCL000822" & cluster == 3 ~ "chondrocyte",
+      # clusters with high "immune" marker gene expression are macrophage
+      library_id == "SCPCL000822" & cluster == 6 ~ "macrophage",
+      # cluster 8 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000822" & cluster == 8 ~ "mesenchymal-like cell",
+      library_id == "SCPCL000824" & cluster %in% c(1, 2, 3, 4, 5, 6, 11) ~ "tumor",
+      library_id == "SCPCL000824" & cluster == 8 ~ "endothelial cell",
+      # cluster 10 has high immune markers and is mostly macrohpages
+      library_id == "SCPCL000824" & cluster == 10 ~ "macrophage",
+      # cluster 7 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000824" & cluster == 7 ~ "mesenchymal-like cell",
+      # if they didn't have high expression of any markers, we label them as unknown
+      .default = "unknown"
+    ),
+    # make tumor first
+    cluster_celltype = forcats::fct_relevel(cluster_celltype, "tumor", after = 0)
+  )
+```
+
+```{r, fig.height=7, fig.width=7}
+# look at cell type assignments on the UMAP
+ggplot(cluster_umap_df, aes(x = UMAP1, y = UMAP2, color = cluster_celltype)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  facet_grid(cols = vars(library_id)) +
+  theme(
+    aspect.ratio = 1,
+    legend.position = "bottom"
+  ) +
+  guides(color = guide_legend(override.aes = list(alpha = 1, size = 1.5))) +
+  scale_color_brewer(palette = "Dark2")
+```
+
+We can also validate that these cells do in fact have higher expression of the marker genes as compared to other cells by plotting the marker gene expression on UMAPs. 
+
+```{r, fig.height=7, fig.width=7}
+# grab columns that contain marker gene sums
+marker_gene_columns <- colnames(cluster_umap_df)[which(endsWith(colnames(cluster_umap_df), "_sum"))]
+
+# for each sample show a faceted umap by cell type
+# color is gene expression of all marker genes for that cell type
+library_ids |>
+  purrr::map(\(id){
+    plot_df <- cluster_umap_df |>
+      dplyr::filter(library_id == id) |>
+      tidyr::pivot_longer(
+        cols = all_of(marker_gene_columns),
+        names_to = "cell_type",
+        values_to = "gene_expression"
+      ) |>
+      dplyr::mutate(cell_type = stringr::str_remove(cell_type, "_sum"))
+
+    ggplot(plot_df, aes(x = UMAP1, y = UMAP2, color = gene_expression)) +
+      geom_point(alpha = 0.5, size = 0.2) +
+      facet_wrap(vars(cell_type)) +
```

Comment:
I wonder if `scales="free"` would help with the color scale, but also it might just do something bonkers, or do nothing at all!...never tried it before for such a circumstance..